examples/standalone/lrauv_control/lrauv_control.cc

```diff
@@ -0,0 +1,219 @@
+      }
+
+      targetSpeed = std::move(_speed);
+      targetYawAngle = std::move(_yaw);
+      targetPitchAngle = std::move(_pitch);
+    }
+
+    // Update the errors after an iteration of control loop.
+    void UpdateErrors()
+    {
+      errorSpeed = targetSpeed - speed;
+      errorSpeedIntegral =
+        std::min(errorSpeedIntegral + errorSpeed, errorSpeedIntegralMax);
+
+      errorYawAngle = targetYawAngle - yawAngle;
+      errorPitchAngle = targetPitchAngle - pitchAngle;
+    }
+
+    // Generate control input to be applied to the thruster.
+    double SpeedControl()
+    {
+      return errorSpeed * kSpeed + errorSpeedIntegral * kISpeed;
+    }
+
+    // Generate control input to be supplied to the yaw rudders.
+    double YawControl()
+    {
+      return errorYawAngle * kYawAngle;
+    }
+
+    // Generate control input to be supplied to the pitch rudders.
+    double PitchControl()
+    {
+      return errorPitchAngle * kPitchAngle;
+    }
+};
+
+int main(int argc, char** argv)
+{
+  auto control = Controller();
+
+  if (argc == 4)
+  {
+    double targetSpeed = std::stod(argv[1]);
+    double targetYaw = std::stod(argv[2]);
+    double targetPitch = std::stod(argv[3]);
+
+    // Target state : speed (m/s), yaw angle, pitch angle (rad).
+    control.SetTargets(targetSpeed ,targetYaw ,targetPitch);
+  }
+
+  transport::Node node;
+
+  // Propeller command publisher.
+  auto propellerTopicTethys =
+    gz::transport::TopicUtils::AsValidTopic(
+      "/model/tethys/joint/propeller_joint/cmd_thrust");
+  auto propellerPubTethys =
+    node.Advertise<gz::msgs::Double>(propellerTopicTethys);
+
+  // Subscriber for vehicle pose.
+  std::function<void(const msgs::Odometry &)> cbPos =
+    [&](const msgs::Odometry &_msg)
+  {
+    auto orientation = _msg.pose().orientation();
+
+    gz::math::Quaterniond q(
+        orientation.w(),
+        orientation.x(),
+        orientation.y(),
+        orientation.z());
+
+    // Update yaw and pitch angles.
+    auto rpy = q.Euler();
+    control.yawAngle = rpy[2];
+    control.pitchAngle = rpy[1];
+
+    gz::math::Vector3d velocity(
+        _msg.twist().linear().x(),
+        _msg.twist().linear().y(),
+        _msg.twist().linear().z());
+
+    control.speed = velocity.Length();
+  };
+
+  node.Subscribe("/model/tethys/odometry", cbPos);
+
+  // Rudder command publisher.
+  auto rudderTopicTethys =
+    gz::transport::TopicUtils::AsValidTopic(
+      "/model/tethys/joint/vertical_fins_joint/0/cmd_pos");
+  auto rudderPubTethys =
+    node.Advertise<gz::msgs::Double>(rudderTopicTethys);
+
+  // Fin command publisher.
+  auto finTopicTethys =
+    gz::transport::TopicUtils::AsValidTopic(
+      "/model/tethys/joint/horizontal_fins_joint/0/cmd_pos");
+  auto finPubTethys =
+    node.Advertise<gz::msgs::Double>(finTopicTethys);
+
+  while (true)
+  {
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+    // Update errors in the controller.
+    control.UpdateErrors();
```

Comment:
@adityapande-1995 meta: if feedback went through a single function (like setpoints do), a separate `UpdateErrors()` call wouldn't be necessary.

Reply:
Done https://github.com/gazebosim/gz-sim/pull/1822/commits/7121710a0b9dc3ada25e55e25a457225e6c75b4d